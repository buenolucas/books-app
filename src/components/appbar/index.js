import MainBar from './MainBar';
import ModalBar from './ModalBar';
import ResultsBar from './ResultsBar';

export {MainBar, ModalBar, ResultsBar};
