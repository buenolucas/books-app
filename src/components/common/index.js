import Button from './Button';
import GoogleBooks from './GoogleBooks';
import Icon from './Icon';
import IconButton from './IconButton';
import RoundedButton from './RoundedButton';
import Typography from './Typography';
import {CheckBox} from './checkbox/';

export {
  Button,
  GoogleBooks,
  Icon,
  IconButton,
  RoundedButton,
  Typography,
  CheckBox,
};
