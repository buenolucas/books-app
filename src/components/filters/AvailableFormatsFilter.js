import React from 'react';
import PropTypes from 'prop-types';
import styled, {withTheme} from 'styled-components/native';
import {CheckBox, Typography} from '../common';

const Container = styled.View``;

const AvailableFormats = ({updateFilter, selectedFilters}) => {
  return (
    <Container>
      <Typography type="title">Formatos Dispon√≠veis</Typography>
      <CheckBox
        label="Epub"
        checked={selectedFilters.indexOf('epub') > -1 && true}
        onPress={() => updateFilter('epub')}
      />
      <CheckBox
        label="PDF"
        checked={selectedFilters.indexOf('pdf') > -1 && true}
        onPress={() => updateFilter('pdf')}
      />
    </Container>
  );
};

export default AvailableFormats;
