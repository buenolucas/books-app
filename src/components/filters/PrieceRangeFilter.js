import React from 'react';
import PropTypes from 'prop-types';
import styled, {withTheme} from 'styled-components/native';
import {CheckBox, Typography} from '../common';

const Container = styled.View``;

const PrieceRangeFilter = ({updateFilter, selectedFilters}) => {
  const filterRange = id => selectedFilters.some(range => range.id === id);

  return (
    <Container>
      <Typography type="title">Preço</Typography>
      <CheckBox
        label="de R$0 até R$30"
        checked={filterRange('range1')}
        onPress={() => updateFilter({id: 'range1', min: 0, max: 30.99})}
      />
      <CheckBox
        label="de R$31 até R$50"
        checked={filterRange('range2')}
        onPress={() => updateFilter({id: 'range2', min: 31, max: 50.99})}
      />
      <CheckBox
        label="de R$51 até R$100"
        checked={filterRange('range3')}
        onPress={() => updateFilter({id: 'range3', min: 51, max: 100.99})}
      />
      <CheckBox
        label="Mais de R$100"
        checked={filterRange('range4')}
        onPress={() => updateFilter({id: 'range4', min: 101, max: Infinity})}
      />
    </Container>
  );
};

PrieceRangeFilter.propTypes = {
  updateFilter: PropTypes.func.isRequired,
  selectedFilters: PropTypes.array.isRequired,
};

export default PrieceRangeFilter;
