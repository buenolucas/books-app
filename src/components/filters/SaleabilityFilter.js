import React from 'react';
import PropTypes from 'prop-types';
import styled, {withTheme} from 'styled-components/native';
import {CheckBox, Typography} from '../common';

const Container = styled.View``;

const SaleabilityFilter = ({updateFilter, selectedFilters}) => {
  return (
    <Container>
      <Typography type="title">Disponibilidade</Typography>
      <CheckBox
        label="Gratuito"
        checked={selectedFilters.indexOf('FREE') > -1 && true}
        onPress={() => updateFilter('FREE')}
      />
      <CheckBox
        label="À Venda"
        checked={selectedFilters.indexOf('FOR_SALE') > -1 && true}
        onPress={() => updateFilter('FOR_SALE')}
      />
      <CheckBox
        label="Pré-Venda"
        checked={selectedFilters.indexOf('FOR_PREORDER') > -1 && true}
        onPress={() => updateFilter('FOR_PREORDER')}
      />
      <CheckBox
        label="Indispoível"
        checked={selectedFilters.indexOf('NOT_FOR_SALE') > -1 && true}
        onPress={() => updateFilter('NOT_FOR_SALE')}
      />
    </Container>
  );
};

SaleabilityFilter.propTypes = {
  updateFilter: PropTypes.func.isRequired,
  selectedFilters: PropTypes.array.isRequired,
};

export default SaleabilityFilter;
