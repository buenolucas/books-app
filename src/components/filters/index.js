import AvailableFormatsFilter from './AvailableFormatsFilter';
import PrieceRangeFilter from './PrieceRangeFilter';
import SaleabilityFilter from './SaleabilityFilter';

export {AvailableFormatsFilter, PrieceRangeFilter, SaleabilityFilter};
