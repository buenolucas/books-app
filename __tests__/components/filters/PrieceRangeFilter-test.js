import React from 'react';
import {shallow} from 'enzyme';

import 'jest-styled-components';
import {PrieceRangeFilter} from '../../../src/components/filters';

const pressFn = jest.fn();

describe('PrieceRangeFilter', () => {
  describe('Rendering', () => {
    it('renders without crashing', () => {
      const filters = [];
      const component = shallow(
        <PrieceRangeFilter selectedFilters={filters} updateFilter={pressFn} />,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
