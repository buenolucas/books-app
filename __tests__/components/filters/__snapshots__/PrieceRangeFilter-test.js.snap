// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrieceRangeFilter Rendering renders without crashing 1`] = `
<Styled(View)>
  <Typography
    primary={true}
    type="title"
  >
    Preço
  </Typography>
  <WithTheme(CheckBox)
    checked={false}
    label="de R$0 até R$30"
    onPress={[Function]}
  />
  <WithTheme(CheckBox)
    checked={false}
    label="de R$31 até R$50"
    onPress={[Function]}
  />
  <WithTheme(CheckBox)
    checked={false}
    label="de R$51 até R$100"
    onPress={[Function]}
  />
  <WithTheme(CheckBox)
    checked={false}
    label="Mais de R$100"
    onPress={[Function]}
  />
</Styled(View)>
`;
